### PR TITLE
Fix navigation aria-label axe accessibility warnings

### PR DESCRIPTION
**Pull request by kicferk1**
### Context
I merged the not-so-accessible change, Callum merged accessibility checks after, and voila, the accessibility tests are now failing on dev.

### Changes proposed in this pull request
Add aria labels to nav items everywhere (in all 2 places)

### Guidance to review
Feel free to suggest better aria labels...

### Testing
It's already tested, and broken.

### CODE REVIEW

**Comment by kicferk1** [comment]
I guess while we are at it, `a11y` runs before our `aaa-meta` test (I think). Is there a less hacky way to tell cypress what order to run things in?

File: spec/cypress/integration/a11y.js [hunk 1]
```diff
@@ -36,7 +36,11 @@ describe("Accessibility", () => {
     cy.checkA11y();
 
     cy.get('[action="/users/sign_in_with_token"] [name="commit"]').click();
-    cy.get("h1").should("contain", "User dashboard");
+    cy.get("@userData").then((user) => {
+      // Update user as previous step caused login token to change
```
Comment (callumacrae) [inline spec/cypress/integration/a11y.js:40]: you weren't messing around when you were talking about copying and pasting test code 🤪

Comment (kicferk1) [reply spec/cypress/integration/a11y.js:40]: ah crap

File: app/views/core_induction_programme/lesson_parts/show.html.erb [hunk 2]
```diff
@@ -54,7 +54,7 @@
       <% end %>
     <% end %>
 
-    <nav class="step-by-step-navigation" role="navigation">
+    <nav class="step-by-step-navigation" role="navigation" aria-label="lesson-part-navigation">
```
Comment (callumacrae) [inline app/views/core_induction_programme/lesson_parts/show.html.erb:57]: What's with the hyphens?

Comment (kicferk1) [reply app/views/core_induction_programme/lesson_parts/show.html.erb:57]: Lack of thought yesterday evening is what it is :P

**Review by callumacrae — APPROVED** [review]

**Pull request merged**